### ISSUE TITLE
Convert backend to TS

### ISSUE THREAD

**Issue by monkeytypegeorge** [Miodec/monkeytype#2430]

**Comment by mrbrianevans** [comment]
I would like to work on this

**Comment by Miodec** [comment]
> I would like to work on this

Awesome. Please start with a pr for compilation setup.

**Comment by Bruception** [comment]
> Okay #2452 opened. Working on it

Can we do this after the backend overhaul is complete?

**Comment by mrbrianevans** [comment]
@Bruception #2452 should really be merged as soon as possible. The longer its left seperate, the more out-of-date it gets. I converted all the `require`s to `imports` etc and successfully got it to compile to typescript and run. If it is merged, and then the backend is overhauled, at least the changes to backend will be using `import` and `export`, rather than just creating more files that need to be migrated. 

Are there any issues that anyone is aware of that are preventing the "Compile backend with Typescript" PR from being merged?

The way its implemented, all the files can stay `.js` for the moment, and still be compiled by typescript, as long as their using ES module import syntax. So type information doesn't need to be immediately added to the whole backend. The PR just adds compilation, but doesn't require files to be converted to typescript.

**Comment by mrbrianevans** [comment]
Converting the backend to TS is now apart of the backend API overhaul. Compilation is set up on the `api-overhaul` branch (#2515). 

Converting any backend files to typescript should be done on the `api-overhaul` branch and NOT `typescript`. The `typescript` branch is for the frontend, and appears to be unmaintained.

(please could someone with permissions update the checklist in the issue description to tick off set up compilation)

**Comment by Miodec** [comment]
> (please could someone with permissions update the checklist in the issue description to tick off set up compilation)

If i assign you to this issue do you then have permission to update the checklist?

**Comment by mrbrianevans** [comment]
nope, still can't edit

**Comment by Miodec** [comment]
Alright, ill mark it as completed then.

**Comment by Bruception** [comment]
> Converting the backend to TS is now apart of the backend API overhaul. Compilation is set up on the `api-overhaul` branch (#2515).
> 
> Converting any backend files to typescript should be done on the `api-overhaul` branch and NOT `typescript`. The `typescript` branch is for the frontend, and appears to be unmaintained.
> 
> (please could someone with permissions update the checklist in the issue description to tick off set up compilation)

I was thinking to continue with the backend TS conversion once API overhaul is merged.

**Comment by mrbrianevans** [comment]
Yeah sure, I just wanted to make sure nobody made the mistake of doing backend typescript on the typescript branch. I agree it makes sense to do one thing at a time and wait for api-overhaul to be merged before converting files to typescript.

**Comment by MathieuGravel** [comment]
@mrbrianevans, I would like to work on this task. I was wondering how does it work, should I write which part I want to work on in the chat since the list can't be edited, and to be sure, pull request are still on the `api-overhaul` branch?

**Comment by Bruception** [comment]
> @mrbrianevans, I would like to work on this task. I was wondering how does it work, should I write which part I want to work on in the chat since the list can't be edited, and to be sure, pull request are still on the `api-overhaul` branch?

No typescript changes are being made at the moment, not until the API backend overhaul is complete. This should be soon though.

**Comment by MathieuGravel** [comment]
> > @mrbrianevans, I would like to work on this task. I was wondering how does it work, should I write which part I want to work on in the chat since the list can't be edited, and to be sure, pull request are still on the `api-overhaul` branch?
> 
> No typescript changes are being made at the moment, not until the API backend overhaul is complete. This should be soon though.

When `api-overhaul` will be merged, every subtask will have their own branches or there will be a branch for all typescript changes. I suppose it will be in this channel that all this information will be post? Thanks for your time, and I will look forward to when changes can be made.

**Comment by mrbrianevans** [comment]
Hi @MathieuGravel , there is a developers discord channel where a lot of discussion takes place. It might be good for you to join it if you want to stay up to date with the typescript conversion.

I think at the moment the plan is to wait for api-overhaul to be merged and then convert to typescript on the master branch?

**Comment by rizwanmustafa** [comment]
Miodec, I want to work on the handlers.
